docs/posts/wow-analysis/index.qmd

```diff
@@ -0,0 +1,174 @@
+
+```{python}
+max_level_chars = wowah_data.filter(_.level == 80).select(_.char).distinct()
+wowah_data_filtered = (
+    wowah_data
+    .join(max_level_chars, "char", how="inner")
+    .filter(_.level.isin([70, 80]))
+    .select(_.char, _.level, _.timestamp)
+)
+
+wowah_data_filtered
+```
+
+```{python}
+level_calc = (
+    wowah_data_filtered.group_by(["char"])
+    .mutate(
+        ts_70=_.timestamp.max(where=_.level == 70),
+        ts_80=_.timestamp.min(where=_.level == 80),
+    )
+    .drop(["level", "timestamp"])
+    .distinct()
+    .mutate(days_from_70_to_80=(_.ts_80.delta(_.ts_70, "day")))
+    .order_by(_.days_from_70_to_80)
+)
+
+level_calc
+```
+The data is filtered and grouped by character, and two new columns are created to represent timestamps for levels 70 and 80. The data is then cleaned, and a new column is created to calculate the time taken to level from 70 to 80. Finally, the dataset is sorted by this column.
+
+This isn’t perfect, as I found a case where there was a player who seemed to have quit in March and then returned for the new expansion. They hit 71 before it looks like their login at 70 was captured later. If you’re curious, take a look at **char=21951** for yourself.
+
+### How did they level?
+
+Let’s grab all the details from the previous result and join it back to get the timestamp and zone data.
+
+```{python}
+leveler_zones = (
+    level_calc.join(wowah_data, "char", how="inner")
+    .filter((_.timestamp >= _.ts_70) & (_.timestamp <= _.ts_80))
+    .group_by([_.char, _.zone])
+    .agg(zone_count=_.zone.count())
+)
+```
+
+This code summarizes how often those characters appear in different zones while leveling up from level 70 to 80. It combines two sets of data based on character names, selects records within the leveling timeframe, groups data by character and zone, and counts the number of times each character was found in each zone.
+
+There is another example table we can join to figure out the Zone information. I’m only interested in two columns, so I’ll filter this further and rename the columns.
+
+```{python}
+zones = ex.wowah_zones_raw.fetch()
+zones = zones.select(zone=_.Zone, zone_type=_.Type)
+zones
+```
+
+Making use of `pivot_wider` and joining back to our `leveler_zones` expression will make this a breeze!
+
+```{python}
+zones_pivot = (
+    leveler_zones.join(zones, "zone")
+    .group_by([_.char, _.zone_type])
+    .agg(zone_type_count=_.zone.count())
+    .pivot_wider(names_from="zone_type", values_from="zone_type_count")
+)
+
+zones_pivot
+```
+
+## Conclusion
+
+Doing complicated analysis tasks with Ibis is pretty straightforward. I find that Ibis makes data preparation much easier, especially when trying to apply bulk operations. Breaking things down into steps with these deferred engines makes things much more manageable.
+
+What other questions could we ask of this data?
```

Comment:
Usually we have a lightweight CTA here (join Zulip, report bug, request features etc), e.g., https://ibis-project.org/posts/duckdb-for-rag/#next-steps.

Up to you! Maybe something like `Get in touch on GitHub, we'd love to see more analyses of this data set`?

Reply:
That is a much better CTA! I will change it to this.

```
## Next Steps

Get in touch with us on [GitHub](https://github.com/ibis-project) or 
[Zulip](https://ibis-project.zulipchat.com/), we'd love to see more analyses of this 
data set. 
```